lib/Pages/otp_verification.dart

```diff
@@ -52,6 +58,93 @@ class _OTPVerificationPageState extends State<OTPVerificationPage> {
     });
   }
 
+  // Method to collect OTP from text fields
+  String _getEnteredOtp() {
+    return _otpControllers.map((controller) => controller.text).join();
+  }
+
+  // Method to verify OTP
+  Future<void> _verifyOtp() async {
+    // Validate OTP length
+    final enteredOtp = _getEnteredOtp();
+    if (enteredOtp.length != 4) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        const SnackBar(content: Text('Please enter a 4-digit OTP')),
+      );
+      return;
+    }
+
+    // Set verifying state
+    setState(() {
+      _isVerifying = true;
+    });
+
+    try {
+      // Call OTP verification service
+      final result = await _otpService.verifyOtp(
+          mobileNumber: widget.phoneNumber,
+          role: UserRole.CUSTOMER,
```

Comment:
this is vendor not customer

Role Mapping
SUPPORT (Role: 1)

ADMIN (Role: 2)

ADMIN_VENDOR_PROVIDER (Role: 3)

VENDOR (Role: 4)

CUSTOMER (Role: 5)